internal/filesystem/file.go

```diff
@@ -40,6 +43,14 @@ func (f *file) URI() string {
 	return URIFromPath(f.fullPath)
 }
 
+func (f *file) Version() int {
+	return f.version
+}
+
+func (f *file) IncrementVersion() {
+	f.version += 1
```

Comment:
according to the lsp spec, one `didchange` event could contain a list of changes, so we should not update the version in every single apply change ?

Reply:
I see - good point - I missed that completely! 🤦 

Just one nitpicky note then - How would you feel about changing the interface to `SetVersion(version int)`, instead of calculating the next version inside, just to ensure that the written version will _always_ match?

Reply:
sure, I will made this change soon

Reply:
I finally choose to invoke the `setVersion` func in the `change` func, because if doing it in the apply change func, we need to add a `version` field for the change object, that seems need to change some other place and test case, I am sure whether it's good to change it